lib/readline.js

```diff
@@ -435,12 +436,21 @@ Interface.prototype._normalWrite = function(b) {
     this._sawReturnAt = string.endsWith('\r') ? Date.now() : 0;
 
     // got one or more newlines; process into "line" events
-    var lines = string.split(lineEnding);
+    var lines = string.split(lineEndingWithSeparator);
     // either '' or (conceivably) the unfinished portion of the next line
     string = lines.pop();
     this._line_buffer = string;
-    for (var n = 0; n < lines.length; n++)
-      this._onLine(lines[n]);
+    for (var n = 0; n < lines.length; n++) {
+      const _line = lines[n];
+      if (lineEndingSeparatorInclusion.test(_line)) {
+        const lineWithoutSeparator = _line.slice(0, -1);
+        const separator = _line.slice(-1);
+
+        this._onLine(lineWithoutSeparator, separator);
+      } else {
+        this._onLine(_line, '\n');
+      }
```

Comment:
If I am not mistaken, each line has a separator here and the else case will not be reached besides potentially for the very last line. Instead of checking with a regular expression what is used, it would be significantly faster to just check the last two characters of each line. But in that case the line would still require to be sliced and that is extra overhead. It's probably faster to use a loop that checks for some special characters and to manually slice reach line.